src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs

```diff
@@ -44,6 +44,117 @@ internal class XplatEventLogger : EventListener
         [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
         private static extern void LogEventSource(int eventID, string? eventName, string eventSourceName, string payload);
 
+        [DllImport(RuntimeHelpers.QCall)]
+        private static extern void LogThreadPoolWorkerThreadStart(uint ActiveWorkerThreadCount, uint RetiredWorkerThreadCount, ushort ClrInstanceID);
+
+        public static void ThreadPoolWorkerThreadStart(uint ActiveWorkerThreadCount, uint RetiredWorkerThreadCount, ushort ClrInstanceID)
+        {
+            LogThreadPoolWorkerThreadStart(ActiveWorkerThreadCount, RetiredWorkerThreadCount, ClrInstanceID);
+        }
```

Comment:
This level of indirection is required?

Reply:
The indirection is mainly because the QCall methods that call into the runtime need to be private, and the public wrappers are for the ThreadPool to be able to call them.

Reply:
Are you sure? Here is an example of a QCall with the internal accesibility modifier: https://github.com/dotnet/runtime/blob/69e114c1abf91241a0eeecf1ecceab4711b8aa62/src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.CoreCLR.cs#L24

I wouldn't have expected QCall to care what accesibility modifier was assigned. If it does need to be non-public internal should still be sufficient to let the intra-assembly calls work without needing the wrapper method.